x/superfluid/keeper/stake_test.go

```diff
@@ -148,6 +151,112 @@ func (suite *KeeperTestSuite) TestSuperfluidDelegate() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestValidateLockForSFDelegate() {
+	lockOwner := suite.TestAccs[0]
+
+	tests := []struct {
+		name                             string
+		lock                             *lockuptypes.PeriodLock
+		sender                           string
+		skParams                         types.Params
+		superfluidAssetToSet             types.SuperfluidAsset
+		lockIdAlreadySuperfluidDelegated bool
+		expectedErr                      error
+	}{
+		{
+			name: "valid gamm lock",
+			lock: &lockuptypes.PeriodLock{
+				Owner:    lockOwner.String(),
+				Coins:    sdk.NewCoins(sdk.NewCoin("gamm/pool/1", sdk.NewInt(100))),
+				Duration: time.Hour * 24 * 21,
+				ID:       1,
+			},
+			superfluidAssetToSet: types.SuperfluidAsset{Denom: "gamm/pool/1", AssetType: types.SuperfluidAssetTypeLPShare},
+			expectedErr:          nil,
+		},
+		{
+			name: "valid cl lock (suffix properly removed)",
+			lock: &lockuptypes.PeriodLock{
+				Owner:    lockOwner.String(),
+				Coins:    sdk.NewCoins(sdk.NewCoin("cl/pool/1/28394022", sdk.NewInt(100))),
+				Duration: time.Hour * 24 * 21,
+				ID:       1,
+			},
+			superfluidAssetToSet: types.SuperfluidAsset{Denom: "cl/pool/1/", AssetType: types.SuperfluidAssetTypeConcentratedShare},
+			expectedErr:          nil,
+		},
+		{
+			name: "invalid lock - not superfluid asset",
+			lock: &lockuptypes.PeriodLock{
+				Owner:    lockOwner.String(),
+				Coins:    sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(100))),
+				Duration: time.Hour * 24 * 21,
+				ID:       1,
+			},
+			superfluidAssetToSet: types.SuperfluidAsset{Denom: "gamm/pool/1", AssetType: types.SuperfluidAssetTypeLPShare},
+			expectedErr:          sdkerrors.Wrapf(types.ErrNonSuperfluidAsset, "denom: %s", "uosmo"),
```

Comment:
nit: would be nice to remove sdkerrors

Reply:
I think this is out of scope, given these sdk errors existed at superfluid launch (agreed on eventually removing all sdk errors though from our repo)